src/main/java/com/ibm/cloud/sdk/core/security/Cp4dToken.java

```diff
@@ -19,21 +19,22 @@
  * This class holds relevant info re: a CP4D access token for use by the CloudPakForDataAuthenticator class.
  */
 public class Cp4dToken extends AbstractToken {
-  public String accessToken;
-  public long expirationTimeInMillis;
+  private String accessToken;
+  private long expirationTimeInMillis;
+  private long refreshTimeInMillis;
 
   /**
    * This ctor is used to store a user-managed access token which will never expire.
    * @param accessToken the user-managed access token
    */
   public Cp4dToken(String accessToken) {
     this.accessToken = accessToken;
-    this.expirationTimeInMillis = -1;
```

Comment:
If you don't set `expirationTimeInMillis` here, I assume it will be zero.  So won't `isTokenValid()` always return `false` for this token?  Is that what we want?

Reply:
Nope, definitely a mistake there. I'll go ahead and set it to -1 like the refresh time to specify that it never expires and update the `isTokenValid()` logic accordingly.